### PR TITLE
Feat: prefill first name in input for events

### PR DESCRIPTION
**Pull request by jsuryakt**
Date: 16 Feb 2024

Developer Name: Jayasurya

---

## Issue Ticket Number:-

#865 

## Description:

Prefill first name of user in events input field

Is Under Feature Flag

- [x] Yes
- [ ] No

Database changes

- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)

- [ ] Yes
- [x] No

Is Development Tested?

- [x] Yes
- [ ] No

Tested in staging?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by iamitprakash — CHANGES_REQUESTED** [review]
test case is missing

**Comment by jsuryakt** [comment]
> test case is missing

There are test cases with name already set, do you mean e2e test?

File: tests/integration/components/live-join-test.js [hunk 1]
```diff
@@ -169,4 +169,40 @@ module('Integration | Component | live-join', function (hooks) {
...
+  test('it should prefill name in input box', async function (assert) {
+    const objToCheckFunctions = {
+      isInputHandler: assert.ok(true, 'inputHandler is working fine!'),
+      isClickHandlerWorks: assert.ok(true, 'clickHandler is working fine!'),
+      isBackHandlerWorks: assert.ok(true, 'backHandler is working fine!'),
+    };
+    this.setProperties({
+      role: 'host',
+      name: 'Jayasurya',
+      roomCode: '',
```
Comment (ajoykumardas12) [inline tests/integration/components/live-join-test.js:182]: This test explicitly sets the name and then checks if name is rendered. Can we instead set login data and check if first name is rendered?

**Comment by satyam73** [comment]
The input can be disabled then if name is already filled

**Comment by jsuryakt** [comment]
> The input can be disabled then if name is already filled

Why? The user can edit it if needed and also if we disable the input, then there is no use of showing it at all, can just proceed with other fields.

File: tests/integration/components/live-join-test.js (hunk 1 above)
Comment (ajoykumardas12) [reply tests/integration/components/live-join-test.js]: Controller name value assertion looks good to me.

**Comment by ajoykumardas12** [comment]
> The input can be disabled then if name is already filled

If not logged in and joining as guest, name won't pre-filled. Not sure if it can/should be disabled when logged in.

**Review by ajoykumardas12 — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**